### PR TITLE
VIM-XXXX: Removes shared client from networking library

### PR DESCRIPTION
**Pull request by rpassis**
#### Ticket

[VIM-XXXX](https://vimean.atlassian.net/browse/VIM-XXXX)

#### Pull Request Checklist

- [X] Resolved any merge conflicts
- [X] No build errors or warnings are introduced
- [X] New files are written in Swift
- [X] New classes contain license headers
- [X] New classes have Documentation
- [X] New public methods have Documentation

#### Issue Summary

Currently the networking library provides a shared singleton instance of `VimeoClient` for consumers to use. We shouldn't make assumption about how API clients will use this library and therefore leave it up to them to decide whether they want to retain a singleton copy of `VimeoClient` or not.

#### Implementation Summary

* Removes convenience `shared` initializer and configure method
* Fixes all unit tests that were making use of the shared singleton in favor of a local client instance.
* Creates a `VimeoClient` factory inside the testing target for convenience
* Removes unused bridging header file from test target

#### How to Test

* Build and run, ensure all tests pass

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 174 tests, with 0 failures (0 unexpected) in 1.111 (1.305) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1397)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Tests/Shared/GCSTests.swift [hunk 1]
```diff
@@ -26,17 +26,13 @@
 
 import XCTest
 import OHHTTPStubs
-import VimeoNetworking
+@testable import VimeoNetworking
 
 class GCSTests: XCTestCase {
+
+
```
Comment (Jumhyn) [inline Tests/Shared/GCSTests.swift]: Nit: extra whitespace

**Review by Jumhyn — APPROVED** [review]
One small nit, take or leave. Otherwise, LGTM

**Comment by rpassis** [comment]
Thank you @Jumhyn!

**Pull request merged**